test/unit/alphabet/composite/alphabet_variant_detail_test.cpp

```diff
@@ -123,3 +124,48 @@ TEST(alphabet_variant_detail_test, char_to_rank)
     EXPECT_EQ(char_to_rank4['G'], 2);
     EXPECT_EQ(char_to_rank4['T'], 4);
 }
+
+TEST(alphabet_variant_detail_test, first_valid_char_table)
+{
+    constexpr std::array first_valid_char_table2 = detail_alphabet_variant<seqan3::dna4,
+                                                                           seqan3::gap>::first_valid_char_table;
+    EXPECT_EQ(first_valid_char_table2.size(), 256u);
+    EXPECT_EQ(first_valid_char_table2['A'], 0);
+    EXPECT_EQ(first_valid_char_table2['C'], 0);
+    EXPECT_EQ(first_valid_char_table2['G'], 0);
+    EXPECT_EQ(first_valid_char_table2['T'], 0);
+    EXPECT_EQ(first_valid_char_table2['-'], 1);
+    EXPECT_EQ(first_valid_char_table2['X'], 2); // invalid char
+
+    constexpr std::array first_valid_char_table3 = detail_alphabet_variant<seqan3::dna4,
+                                                                           seqan3::gap,
+                                                                           seqan3::dna5>::first_valid_char_table;
+    EXPECT_EQ(first_valid_char_table3.size(), 256u);
+    EXPECT_EQ(first_valid_char_table3['A'], 0);
+    EXPECT_EQ(first_valid_char_table3['C'], 0);
+    EXPECT_EQ(first_valid_char_table3['G'], 0);
+    EXPECT_EQ(first_valid_char_table3['T'], 0);
+    EXPECT_EQ(first_valid_char_table3['-'], 1);
+    EXPECT_EQ(first_valid_char_table3['A'], 0);
+    EXPECT_EQ(first_valid_char_table3['C'], 0);
+    EXPECT_EQ(first_valid_char_table3['G'], 0);
+    EXPECT_EQ(first_valid_char_table3['N'], 2);
+    EXPECT_EQ(first_valid_char_table3['T'], 0);
+    EXPECT_EQ(first_valid_char_table3['X'], 3); // invalid char
+
+    constexpr std::array first_valid_char_table4 = detail_alphabet_variant<seqan3::dna5,
+                                                                           seqan3::gap,
+                                                                           seqan3::dna4>::first_valid_char_table;
```

Comment:
Maybe its nicer to check with the following table? Because than you get 'N' with 2?
```suggestion
    constexpr std::array first_valid_char_table4 = detail_alphabet_variant<seqan3::dna4,
                                                                           seqan3::gap,
                                                                           seqan3::dna5>::first_valid_char_table;
```

Reply:
That is already tested by `first_valid_char_table3`